### ISSUE TITLE
Plotting quantile losses over number of runs/epochs and finding prediction width

### ISSUE THREAD

**Issue by parimuns** [awslabs/gluon-ts#1124]
Hello, I am using DeepAR for forecasting and **I want to plot the losses over the number of iterations or epochs**. I am attaching my code below. Also, how can I get the **prediction width of the quantiles** (prediction interval normalised average width).

I have checked earlier issues and could not find a similar issue, so please let me know if earlier it has been answered.
My code snippet is attached below :
[sample.zip](https://github.com/awslabs/gluon-ts/files/5468531/sample.zip)

``` !pip install mxnet
!pip install gluonts==0.5.2
import mxnet as mx
import gluonts
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import json
import os
from itertools import islice
from pathlib import Path

from gluonts.dataset.field_names import FieldName

[f"FieldName.{k} = '{v}'" for k, v in FieldName.__dict__.items() if not k.startswith('_')]



load_train = pd.read_csv(r'sample.csv',parse_dates=[['Date', 'Time']],header=0,index_col=0)
load_train.head()

train=load_train.transpose()
train.head()

from gluonts.dataset.common import ListDataset

train2=train.to_numpy()
type(train2)
train2.shape

feat_static_cat=train2[:,0]
feat_static_cat.shape

target=train2
target.shape

freq='1H'
prediction_length=24
start= [pd.Timestamp("2012-01-01", freq='1H') 
                                for _ in range(22)]


train_ds = ListDataset([{FieldName.TARGET: target, 
                         FieldName.START: start,
                          FieldName.FEAT_STATIC_CAT: [fsc]}
                        for (target, start,fsc) in zip(target[:, :-prediction_length], 
                                                            start,
                                                        feat_static_cat
                                                          )],
                      freq=freq)

from gluonts.model.deepar import DeepAREstimator
from gluonts.trainer import Trainer
from gluonts.distribution import MultivariateGaussianOutput
from gluonts.distribution import NegativeBinomialOutput

from gluonts.time_feature import HourOfDay
t2=HourOfDay(True)

estimator=DeepAREstimator(freq='H',
        prediction_length=24,
        context_length=24,
        cardinality=[22],
        distr_output = NegativeBinomialOutput(),
        scaling=True,
        trainer = Trainer(
            epochs=1, num_batches_per_epoch=32, hybridize=False),
        num_layers= 2,
        num_cells= 40,
        cell_type= "lstm",
        num_parallel_samples = 100,
        dropout_rate= 0.1,
        use_feat_dynamic_real = False,
        use_feat_static_cat = True,
        time_features=[t2]
    )

predictor = estimator.train(train_ds)

from gluonts.evaluation.backtest import make_evaluation_predictions

test_ds = ListDataset([{FieldName.TARGET: target, 
                        FieldName.START: start,
                        # FieldName.FEAT_DYNAMIC_REAL: [fdr],
                        FieldName.FEAT_STATIC_CAT: [fsc]} 
                       for (target, start,fsc) in zip(target, 
                                                            start, 
                                                            # feat_dynamic_real, 
                                                            feat_static_cat)],
                     freq=freq)


len(test_ds)

forecast_it, ts_it = make_evaluation_predictions(
    dataset=test_ds,  # test dataset
    predictor=predictor,  # predictor
    num_samples=100,  # number of sample paths we want for evaluation
)

forecasts = list(forecast_it)
tss = list(ts_it)

def plot_prob_forecasts(ts_entry, forecast_entry):
    plot_length = 150
    prediction_intervals = (90.0, 95.0)
    legend = ["observations", "median prediction"] + [f"{k}% prediction interval" for k in prediction_intervals][::-1]

    fig, ax = plt.subplots(1, 1, figsize=(10, 7))
    ts_entry[-plot_length:].plot(ax=ax)  # plot the time series
    forecast_entry.plot(prediction_intervals=prediction_intervals, color='r')
    plt.grid(which="both")
    plt.legend(legend, loc="upper left")
    plt.show()

plot_prob_forecasts(tss[0], forecasts[0])

from gluonts.evaluation import Evaluator

from gluonts.evaluation import MultivariateEvaluator

len(test_ds)

forecast_it, ts_it = make_evaluation_predictions(
    dataset=test_ds,  
    predictor=predictor,  
    num_samples=100
    )

quantile_seq = np.arange(0.01, 0.99 , .01)
evaluator = Evaluator(quantiles=quantile_seq)
agg_metrics, item_metrics = evaluator(iter(tss), iter(forecasts), num_series=len(test_ds))
